examples/alter_group_offsets.c

```diff
@@ -0,0 +1,336 @@
+
+
+#define fatal(...)                                                             \
+        do {                                                                   \
+                fprintf(stderr, "ERROR: ");                                    \
+                fprintf(stderr, __VA_ARGS__);                                  \
+                fprintf(stderr, "\n");                                         \
+                exit(2);                                                       \
+        } while (0)
+
+
+/**
+ * @brief Set config property. Exit on failure.
+ */
+static void conf_set(rd_kafka_conf_t *conf, const char *name, const char *val) {
+        char errstr[512];
+
+        if (rd_kafka_conf_set(conf, name, val, errstr, sizeof(errstr)) !=
+            RD_KAFKA_CONF_OK)
+                fatal("Failed to set %s=%s: %s", name, val, errstr);
+}
+
+
+static void
+print_partition_list(FILE *fp,
+                     const rd_kafka_topic_partition_list_t *partitions,
+                     int print_offset,
+                     const char *prefix) {
+        if (partitions->cnt == 0) {
+                fprintf(fp, "%sNo partition found", prefix);
+        }
+        int i;
+        char offset_string[521] = {};
+        for (i = 0; i < partitions->cnt; i++) {
+                memset(&offset_string, 0, sizeof(offset_string));
+                if (print_offset) {
+                        snprintf(offset_string, sizeof(offset_string),
+                                 " offset %" PRId64,
+                                 partitions->elems[i].offset);
+                }
+                fprintf(fp, "%s%s %s [%" PRId32 "]%s error %s",
+                        i > 0 ? "\n" : "", prefix, partitions->elems[i].topic,
+                        partitions->elems[i].partition, offset_string,
+                        rd_kafka_err2str(partitions->elems[i].err));
+        }
+        fprintf(fp, "\n");
+}
+
+/**
+ * @brief Parse an integer or fail.
+ */
+int64_t parse_int(const char *what, const char *str) {
+        char *end;
+        unsigned long n = strtoull(str, &end, 0);
+
+        if (end != str + strlen(str)) {
+                fprintf(stderr, "%% Invalid input for %s: %s: not an integer\n",
+                        what, str);
+                exit(1);
+        }
+
+        return (int64_t)n;
+}
+
+static void
+cmd_alter_group_offsets(rd_kafka_conf_t *conf, int argc, char **argv) {
+        char errstr[512]; /* librdkafka API error reporting buffer */
+        rd_kafka_t *rk;   /* Admin client instance */
+        rd_kafka_AdminOptions_t *options; /* (Optional) Options for
+                                           * AlterConsumerGroupOffsets() */
+        rd_kafka_event_t *event; /* AlterConsumerGroupOffsets result event */
+        const int min_argc = 2;
+        int num_partitions = 0;
+
+        /*
+         * Argument validation
+         */
+        int print_usage = argc < min_argc;
+        print_usage |= (argc - min_argc) % 2 != 0;
```

Comment:
since print_usage is only used once, why not make this an easier to read if statement with an or?
```c
if (argc < min_argc || (argc - min_argc) % 2 != 0)
    print_usage("...");
```

Reply:
Done